### ISSUE TITLE
compilation failure for linux x32 ABI

### ISSUE THREAD

**Issue by jpalus** [sudo-project/sudo#158]
sudo 1.9.11 fails to compile while building for x32 ABI target:
```
./exec_ptrace.c: In function 'ptrace_intercept_execve':
./exec_ptrace.c:1039:9: error: duplicate case value
 1039 |         case __NR_execve:
      |         ^~~~
./exec_ptrace.c:1037:9: note: previously used here
 1037 |         case X32_execve:
      |         ^~~~
./exec_ptrace.c:1045:9: error: duplicate case value
 1045 |         case __NR_execveat:
      |         ^~~~
./exec_ptrace.c:1043:9: note: previously used here
 1043 |         case X32_execveat:
      |         ^~~~
```
Seems like x86_64 has support for intercepting x32 binaries but it doesn't distinguish between `__x86_64__` with and without `__ILP32__` leading to attempt of adding support for adding interception of x32 binaries on x32 while I guess it should try adding support for interception of x86_64 binaries on x32.

**Comment by millert** [comment]
This should be fixed by [dfee181](https://github.com/sudo-project/sudo/commit/dfee181d1579a24c679eff960c85bc659c56993f), can you confirm?

**Comment by jpalus** [comment]
Works fine, thanks!
